app/services/authn_context_resolver.rb

```diff
@@ -54,18 +67,96 @@ def no_identity_proofing_vot
     end
   end
 
+  def acr_result
+    @acr_result ||= decorate_acr_result_with_user_context(
+      acr_result_with_sp_defaults,
+    )
+  end
+
   def acr_result_with_sp_defaults
+    @acr_result_with_sp_defaults ||= decorate_acr_result_with_sp_defaults(
+      acr_result_without_sp_defaults,
+    )
+  end
+
+  # @param [Vot::Parser::Result] result Parser result to decorate.
+  # @return [Vot::Parser::Result]
+  def decorate_acr_result_with_sp_defaults(result)
     result_with_sp_aal_defaults(
       result_with_sp_ial_defaults(
-        acr_result_without_sp_defaults,
+        result,
       ),
     )
   end
 
+  # With a given {User}, assert the resolved IAL ACR value and corollary requirements
+  # @param [Vot::Parser::Result] result Parser result to decorate.
+  # @return [Vot::Parser::Result]
+  def decorate_acr_result_with_user_context(result)
+    result_decorator = self.method(:no_identity_proofing_acr_result)
```

Comment:
this method has gotten a lot more complicated with the inclusion of the meta-programming. i understand the desire to be explicit about what's in the result, but this has become harder for me to reason about in the process. 

my personal preference would be closer to my original implementation (where we were just changing the values that needed to be changed, since we know, for instance, if the result is asking for biometric_comparison and it's required, we are not doing any data transformation on the result).

however, if you want to avoid making assumptions/keeping it explicit (which is fair), i would remove the self.method call and just call the methods directly. ruby automatically returns the last statement, so the correct `result` will be returned. the final line could be `no_identity_proofing_acr_result(result)` in order to capture any cases not found. this will make this method more direct and easier to reason about (imo).

Reply:
other things i personally might consider is pulling all the `asserted_ial_acr` logic into a separate decorator (`decorate_result_with_ial_acr`), but that may be an optimization for another day.

Reply:
I agree with Davida that the meta-programming may be a bit much for this implementation.

Reply:
Re: meta-programming. That's fair, I was trying to solve for not having a million base case statements returning the IAL1 acr result. I'll try a different approach.

I don't know that it would be feasible to move the `asserted_ial_acr` logic separate from the updating the result attributes as they are tightly coupled.

Reply:
i had a thought: this `asserted_ial_acr` is explicilty for dermining the return ial value. 

for the purposes of actually answering the question "what is `info[:ial]`", we'll never be answering that question unless there is no identity profile. which means we should only be figuring out what the `asserted_ial_acr` if it's identity proofed. so i think we should be able to make this a lot less complicted knowing that use case?

Reply:
I had forgotten while in the weeds of the code that the vtr param is used as a logic gate. A number of specs were failing because I wasn't looking at the full resolution graph of for biometric proofing, but rather only places where AuthnContextResolver#resolve and Vot::Parser::Result were being directly utilized.
I changed the behavior of the resolver with that narrow focus in mind, which meant absorbing some business logic around IAL level assertion. 

I'm going to unpublish the PR and attempt to streamline the changes further.